goSrc/src/vmCaller/rpc/eth_service.go

```diff
@@ -0,0 +1,1034 @@
+		}
+		tx.Address = &addr
+	}
+
+	// gas provided for the transaction execution
+	if gasLimit := req.Transaction.Gas; gasLimit != "" {
+		tx.GasLimit, err = strconv.ParseUint(gasLimit, 0, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse gasLimit: %v", err)
+		}
+	}
+
+	if gasPrice := req.Transaction.GasPrice; gasPrice != "" {
+		tx.GasPrice, err = strconv.ParseUint(gasPrice, 0, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse gasPrice: %v", err)
+		}
+	}
+
+	if data := req.Transaction.Data; data != "" {
+		bs, err := x.DecodeToBytes(data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse data: %v", err)
+		}
+		tx.Data = bin.HexBytes(bs)
+	}
+
+	txEnv := txs.Enclose(srv.blockchain.ChainID(), tx)
+
+	signer, err := keys.AddressableSigner(srv.keyClient, tx.Input.Address)
+	if err != nil {
+		return nil, err
+	}
+	err = txEnv.Sign(signer)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	txe, err := srv.trans.BroadcastTxSync(ctx, txEnv)
+	if err != nil {
+		return nil, err
+	} else if txe.Exception != nil {
+		return nil, txe.Exception.AsError()
+	}
+
+	return &web3.EthSendTransactionResult{
+		TransactionHash: x.EncodeBytes(txe.GetTxHash().Bytes()),
+	}, nil
+}
+
+// EthAccounts returns all accounts signable from the local node
+func (srv *EthService) EthAccounts() (*web3.EthAccountsResult, error) {
+	addresses, err := srv.keyStore.GetAllAddresses()
+	if err != nil {
+		return nil, err
+	}
+
+	addrs := make([]string, 0, len(addresses))
+	for _, addr := range addresses {
+		data, err := hex.DecodeString(addr)
+		if err != nil {
+			return nil, fmt.Errorf("could not decode address %s", addr)
+		}
+		key, err := srv.keyStore.GetKey("", data)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve key for %s", addr)
+		} else if key.CurveType != crypto.CurveTypeSecp256k1 {
+			// we only want ethereum keys
+			continue
+		}
+		// TODO: only return accounts that exist in current chain
+		addrs = append(addrs, x.EncodeBytes(key.Address.Bytes()))
+	}
+
+	return &web3.EthAccountsResult{
+		Addresses: addrs,
+	}, nil
+}
+
+// EthSign: https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign
+func (srv *EthService) EthSign(req *web3.EthSignParams) (*web3.EthSignResult, error) {
+	addr, err := x.DecodeToBytes(req.Address)
+	if err != nil {
+		return nil, err
+	}
+	to, err := crypto.AddressFromBytes(addr)
+	if err != nil {
+		return nil, err
+	}
+	signer, err := keys.AddressableSigner(srv.keyClient, to)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := x.DecodeToBytes(req.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := append([]byte{0x19}, []byte("Ethereum Signed Message:\n")...)
+	msg = append(msg, byte(len(data)))
+	msg = append(msg, data...)
+
+	sig, err := signer.Sign(crypto.Keccak256(msg))
+	if err != nil {
+		return nil, err
+	}
+
+	return &web3.EthSignResult{
+		Signature: x.EncodeBytes(sig.RawBytes()),
+	}, nil
+}
+
+// N / A
```

Comment:
?

Reply:
not added, also comment from burrow team